nodemanagement/node_management_worker.go

```diff
@@ -201,22 +201,32 @@ func (n *NodeManagementWorker) DownloadComplete(cmd *NMPDownloadCompleteCommand)
 	}
 	var msgMeta *persistence.MessageMeta
 	eventCode := ""
-	if cmd.Msg.Status == exchangecommon.STATUS_DOWNLOADED {
-		glog.Infof(nmwlog(fmt.Sprintf("Sucessfully downloaded packages for nmp %v.", cmd.Msg.NMPName)))
-		status.SetStatus(exchangecommon.STATUS_DOWNLOADED)
-		msgMeta = persistence.NewMessageMeta(EL_NMP_STATUS_CHANGED, cmd.Msg.NMPName, exchangecommon.STATUS_DOWNLOADED)
-		eventCode = persistence.EC_NMP_STATUS_DOWNLOAD_SUCCESSFUL
-	} else if cmd.Msg.Status == exchangecommon.STATUS_NO_ACTION {
+
+	if cmd.Msg.Status == exchangecommon.STATUS_NO_ACTION {
 		glog.Infof(nmwlog(fmt.Sprintf("Already in compliance with nmp %v. Download skipped.", cmd.Msg.NMPName)))
 		status.SetStatus(exchangecommon.STATUS_NO_ACTION)
 		msgMeta = persistence.NewMessageMeta(EL_NMP_STATUS_CHANGED, cmd.Msg.NMPName, exchangecommon.STATUS_NO_ACTION)
 		eventCode = persistence.EC_NMP_STATUS_DOWNLOAD_SUCCESSFUL
+	} else 	if cmd.Msg.Status == exchangecommon.STATUS_DOWNLOADED {
+		glog.Infof(nmwlog(fmt.Sprintf("Sucessfully downloaded packages for nmp %v.", cmd.Msg.NMPName)))
+		status.SetStatus(exchangecommon.STATUS_DOWNLOADED)
+		msgMeta = persistence.NewMessageMeta(EL_NMP_STATUS_CHANGED, cmd.Msg.NMPName, exchangecommon.STATUS_DOWNLOADED)
+		eventCode = persistence.EC_NMP_STATUS_DOWNLOAD_SUCCESSFUL
 	} else {
```

Comment:
I have added `STATUS_PRECHECK_FAILED` for function download_worker.DownloadAgentUpgradePackages. Please check this status too and save the status.